open_optical_gating/cli/determine_reference_period.py

```diff
@@ -180,6 +224,8 @@ def gotScoreForDelta(score, d, values):
 
 
 def save_period(reference_period, period_dir="~/"):
+    # TODO: JT writes: Needs an explanation of parameters and purpose, and what the isinstance test is about
```

Comment:
Wow.... er.... got no idea about this isinstance....... I guess at some point I accidentally passed it an int?

Reply:
What's your preferred solution? I'm not a fan of having extra lines of code that exist just to catch (or rather, ignore!) misuse like this. I would either add an assert(*), or just remove the test entirely. 

(*) I have a feeling you're averse to non-critical asserts. This is in a sense non-critical, in that it can be ignored, but I'm not a fan of brushing things under the carpet...

Reply:
I think we should just delete it... it seems like a nonsense check to me.

(*) asserts always seem weird to me, if I want to break my code under certain conditions then I would raise an exception, if I want to check things but not break my code I would log, which is what I would recommend if we are going to keep this catch in. My general rule of thumb for the different log levels is:
- `trace` for 'stuff that I may need to see but is impossibly to print cleanly,' e.g. large arrays that would just fill up my terminal
- `debug` for 'stuff I will want to print out if something breaks`
- `info` for 'expected breaks with no consequences`, in the sense that the code is working (so it's a success) but whats working is that something isn't working (so not a success), e.g. logger.info('No object found but I can carry on find so this is just info.')
- `success` for 'this worked and most users will want to know about it', these are sort of things that non-devs will want to know, e.g. 'i connected to the stages fine'
- `warning` for 'expected breaks with possible consequences', so like info except this could cause bad results or a later failure
- `error` for unexpected breaks, so when something the user requests happen can't, e.g. the user says move the stage but the code can't
- `critical` for failure, e.g. the user says 'get me a stage controller' and the code can't - this will break all future requests to do with the stage so it is critical and not just an error.